src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,567 +51,718 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+                !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update)) {
+                    updateWrapper = new HandleUpdateWrapperLogic (c, oldContacts, triggerAction, i);
+                }
+            i += 1;
+        }
+
+        //Set recursion flag
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert, true);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update, true);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterDelete) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Delete, true);
+        }
+
+        handleInsertProcessing(insertWrapper, contactsNeedAccounts);
+        handleUpdateProcessing(updateWrapper, contactsNeedAccounts);
+    }
+
+    /*******************************************************************************************************
+    * @description Processes after insert logic.
+    * @return HandleInsertWrapperLogic.
+    */
+    private class HandleInsertWrapperLogic {
+        public Map<Id,Id> accIdConIdToUpdate = new Map<Id,Id>();
+        public Set<Id> accountIdToRename = new Set<Id>();
+
+        private HandleInsertWrapperLogic(Contact c) {
+            if (c.AccountId != NULL) {
+                //(1) If the Contact is connected to an Account but Primary_Contact__c on Account is NULL
+                if (c.Account.Primary_Contact__c == NULL) {
+                    accIdConIdToUpdate.put(c.AccountId, c.Id);
+                }
+
+                //(2) If Contacts are connected to Accounts, make the association in the other direction
+                if (checkHouseholdNaming(c.Account.RecordTypeID)) {
+                    accountIdToRename.add(c.AccountId);
                 }
             }
-        } else {
-            Integer i = 0; // used to index into oldContacts when needed
-            for (Contact c : contacts) {
+        }
+    }
 
-                // AFTER INSERT
-                if (triggerAction == TDTM_Runnable.Action.AfterInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert)) {
+    /*******************************************************************************************************
+    * @description This method checks if any of the after insert collections are not empty.
+    * If not empty, it calls the appropriate methods to process the list/set/map.
+    * @param insertWrapper contains the collections processed from after insert context.
+    * @param contactsNeedAccounts a list of Contacts that needs new Accounts or an update.
+    * @return void.
```

Comment:
Same nitpick as before - anywhere there's no return, we don't need to add it to the docs here.